fiddler/datadog_checks/fiddler/check.py

```diff
@@ -0,0 +1,152 @@
+
+client = fdl.FiddlerApi(url=URL, org_id=ORG_ID, auth_token=AUTH_TOKEN)
+
+
+def create_tags(**tags_in):
+    tags.clear()
+    tags.append("project:" + tags_in['project'])
+    tags.append("model:" + tags_in['model'])
+    if len(tags_in) > 2:
+        tags.append("feature:" + tags_in['feature'])
+
+
+class FiddlerCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'fiddler'
+
+    def __init__(self, name, init_config, instances):
+        super(FiddlerCheck, self).__init__(name, init_config, instances)
+
+    #        self.base_url = self.instance.get('url')
+    #        self.api_key = self.instance.get('fiddler_api_key')
+    #        self.org = self.instance.get('organization')
+
+    #        print("Connecting to : ", self.base_url)
+    #        print("with org id : ", self.org)
+    #        print("and auth key : ", self.api_key)
+
+    #        client = fdl.FiddlerApi(url=self.base_url, org_id=self.org, auth_token=self.api_key)
+
+    # Use self.instance to read the check configuration
+
+    def check(self, _):
+        # Iterate through the projects and the models and push data into Fiddler
+        project_path = ['list_projects', ORG_ID]
+        result_all = client._call(project_path)
+        print("Projects: ", result_all["projects"])
+
+        start_time = (time.time() * 1000) - (bin_size * 1000)
+        end_time = time.time() * 1000
+        print("Start time is : ", start_time)
+        print("End time is : ", end_time)
+
+        # Iterate through all of the projects within the Fiddler instance and get the metrics
+        for project in result_all["projects"]:
+            models = project["models"]
+
+            # Iterate through all of the models within a project
+            for model in models:
+                print("Model: ", model["id"])
+                for metric in metrics_list:
+                    print("Metric is :", metric)
+                    json_request = {
+                        "metric": metric,
+                        "time_range_start": start_time,
+                        "time_range_end": end_time,
+                        "bin_size": bin_size,
+                        "prediction": '_',
+                    }
+                    agg_metrics_path = ['aggregated_metrics', ORG_ID, project["name"], model["id"]]
+                    print("ProjectModel: ", project["name"], model["id"], metric)
+                    # result = client._call(agg_metrics_path, json_request)
+
+                    try:
+                        result = client._call(agg_metrics_path, json_request)
+                    except:
+                        print("Aggregated metrics exception : ", agg_metrics_path)
+                        print("Project with no monitoring data. ProjectModel: ", project["name"], model["id"], metric)
+                        hit_exception = True
+
+                    if hit_exception:
```

Comment:
If there's no exceptions line 90, this variable is not defined. This is one of the reasons why the CI is failing right now.

Reply:
I was missing initializing that variable. Have made that change.